lib/db/schema/patch-023-024.sql

```diff
@@ -0,0 +1,175 @@
+-- Add table for storing unverified sessionTokens.
+CREATE TABLE `unverifiedTokens` (
+  tokenId BINARY(32) NOT NULL PRIMARY KEY,
+  tokenVerificationId BINARY(16) NOT NULL,
+  uid BINARY(16) NOT NULL
+) ENGINE=InnoDB;
+
+-- Update createSessionToken stored procedure to
+-- insert into unverifiedTokens.
+CREATE PROCEDURE `createSessionToken_3` (
+  IN tokenId BINARY(32),
+  IN tokenData BINARY(32),
+  IN uid BINARY(16),
+  IN createdAt BIGINT UNSIGNED,
+  IN uaBrowser VARCHAR(255),
+  IN uaBrowserVersion VARCHAR(255),
+  IN uaOS VARCHAR(255),
+  IN uaOSVersion VARCHAR(255),
+  IN uaDeviceType VARCHAR(255),
+  IN tokenVerificationId BINARY(16)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  INSERT INTO sessionTokens(
+    tokenId,
+    tokenData,
+    uid,
+    createdAt,
+    uaBrowser,
+    uaBrowserVersion,
+    uaOS,
+    uaOSVersion,
+    uaDeviceType,
+    lastAccessTime
+  )
+  VALUES(
+    tokenId,
+    tokenData,
+    uid,
+    createdAt,
+    uaBrowser,
+    uaBrowserVersion,
+    uaOS,
+    uaOSVersion,
+    uaDeviceType,
+    createdAt
+  );
+
+  IF tokenVerificationId IS NOT NULL THEN
+    INSERT INTO unverifiedTokens(
+      tokenId,
+      tokenVerificationId,
+      uid
+    )
+    VALUES(
+      tokenId,
+      tokenVerificationId,
+      uid
+    );
+  END IF;
+
+  COMMIT;
+END;
+
+-- Update deleteSessionToken stored procedure to
+-- delete from unverifiedTokens.
+CREATE PROCEDURE `deleteSessionToken_2` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM sessionTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM unverifiedTokens WHERE tokenId = tokenIdArg;
+
+  COMMIT;
+END;
```

Comment:
The following stored procedures also delete from the `sessionTokens` table, and so need to be updated to delete from `unverifiedTokens` as well:

* deleteDevice
* deleteAccount
* resetAccount